test/keccak-transcript-tests.t.sol

```diff
@@ -81,7 +81,9 @@ contract KeccakTranscriptContractTest is Test {
         assertEq(scalarVesta, expectedVesta);
     }
 
-    function testKeccakTranscriptPallas() public {
+    // Following test has been ported: https://github.com/lurk-lab/Nova/blob/solidity-verifier-pp-spartan/src/provider/keccak.rs#L138
+    // TODO: For some reason, test vector passes if using Vesta curve parameters, but in reference implementation type of point is Pallas
```

Comment:
Is the TODO still necessary, or did we figure out the Pallas <-> Vesta mixup?

Reply:
I think it is still not 100% clear. I mean [following](https://github.com/lurk-lab/Nova/blob/solidity-verifier-pp-spartan/src/provider/keccak.rs#L138) test in reference is parameterised by Pallas point. But given test vector outcome is obtained when we use different curve (Vesta) parameterisation at Solidity side. May be my understanding of reference is not 100% correct... @huitseeker, any thoughts?

Reply:
I think the situation is a bit confusing for the following reason:
- when *Rust* tests are parametrized by a point, it can mean that we are using the *G* parameter because a/ we're indeed digesting a point OR b/ we're using that parameter to link to its associated types, such as `G::Scalar`,
- when we digest a point in the Nova protocol we are in general calling the function `to_coordinates` from the trait `Group`, which returns the components of the affine representation of the point. As specified, those coordinates are members of the base field of the point:
https://github.com/microsoft/Nova/blob/200b47edab190c06a44cbd87a63d34a444184a41/src/traits/mod.rs#L79
In the case of a Pallas Point, this will be two coordinates in the Vesta scalar field.
- often in tests, we also absorb points from `G::Scalar`, which means field elements in the point's own associated scalar field. In the case of a Pallas Point, that's a field element from the Pallas scalar field.